src/KawaiiEngine/src/EventProvider.cpp

```diff
@@ -7,78 +7,54 @@
 
 kawe::EventProvider *kawe::EventProvider::s_instance = nullptr;
 
-kawe::EventProvider::EventProvider() { s_instance = this; }
-
-auto kawe::EventProvider::assign(const Window &window) -> void
+kawe::EventProvider::EventProvider(const Window &window) : m_window{window}
 {
-    ::glfwSetWindowCloseCallback(window.get(), callback_eventClose);
-    ::glfwSetWindowSizeCallback(window.get(), callback_eventResized);
-    ::glfwSetWindowPosCallback(window.get(), callback_eventMoved);
-    ::glfwSetKeyCallback(window.get(), callback_eventKeyBoard);
-    ::glfwSetMouseButtonCallback(window.get(), callback_eventMousePressed);
-    ::glfwSetCursorPosCallback(window.get(), callback_eventMouseMoved);
-    ::glfwSetCharCallback(window.get(), callback_char);
-
-    // todo : ImGui_ImplGlfw_ScrollCallback
-    // glfwSetScrollCallback(window, scroll_callback);
-    // void scroll_callback(GLFWwindow* window, double xoffset, double yoffset)
-
-    // glfwSetJoystickCallback(joystick_configuration_change_handler);
-    // glfwSetFramebufferSizeCallback(window, framebuffer_size_handler);
-    // glfwSetWindowIconifyCallback(window, window_iconify_handler);
-    // glfwSetWindowFocusCallback(window, window_focus_handler);
-    // glfwSetErrorCallback(error_handler);
-
-    m_buffer_events.emplace_back(OpenWindow{});
-    m_events_processed.emplace_back(TimeElapsed{});
+    s_instance = this;
+    setState(State::RECORD);
+    m_buffer_events.emplace_back(event::Connected<event::Window>{});
 }
 
-auto kawe::EventProvider::getNextEvent() -> Event
+auto kawe::EventProvider::getNextEvent() -> event::Event
 {
     const auto event = fetchEvent();
 
-    std::visit(
-        overloaded{
-            [](TimeElapsed &prev, const TimeElapsed &next) { prev.elapsed += next.elapsed; },
-            [&](const auto &, const std::monostate &) {},
-            [&](const auto &, const auto &next) { m_events_processed.push_back(next); }},
-        m_events_processed.back(),
-        event);
+    if (!m_events_processed.empty()) {
+        // post processing to merge the event for example
+        std::visit(
+            overloaded{// [](event::TimeElapsed &prev, const event::TimeElapsed &next) { prev += next; },
```

Comment:
Could be restored by using some kind of stack incrementing and computing the average